src/pal/pal_apple.h

```diff
@@ -50,6 +52,40 @@ namespace snmalloc
       else
         ::bzero(p, size);
     }
+
+    /**
+     * Overriding here to mark the page as reusable
+     * rolling it as much as necessary.
+     * As above, the x86 h/w worked alright without this change
+     * however now large allocations work better and more reliably
+     * with on ARM, not to mention better RSS number accuracy
+     * for tools based on task_info API.
+     */
+    static void notify_not_using(void* p, size_t size) noexcept
+    {
+      SNMALLOC_ASSERT(is_aligned_block<PALBSD::page_size>(p, size));
+      while (madvise(p, size, MADV_FREE_REUSABLE) == -1 && errno == EAGAIN)
```

Comment:
Does this need to be guarded by the same lock at in `notify_using`?

Reply:
yes in my local rebase it got lost indeed.

Reply:
Why does this need protecting by a lock?  Do concurrent calls to this `madvise` call fail?

Reply:
No it does not. As said I was prudent, maybe a bit too much finally. I can remove those if you wish ?

Reply:
If the call won't fail due to concurrent uses, I don't think there's any point in protecting it with a lock.